memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateConversationRequest.java

```diff
@@ -137,12 +137,27 @@ public static CreateConversationRequest fromRestRequest(RestRequest restRequest)
         }
         try (XContentParser parser = restRequest.contentParser()) {
             Map<String, Object> body = parser.map();
+            String name = null;
+            String applicationType = null;
+            Map<String, String> additionalInfo = null;
+
+            for (String key : body.keySet()) {
```

Comment:
In a map why do we need  a loop? Can't we do:

```
// Extract fields with null safety
        String name = (String) body.getOrDefault(ActionConstants.REQUEST_CONVERSATION_NAME_FIELD, null);
        String applicationType = (String) body.getOrDefault(APPLICATION_TYPE_FIELD, null);
        Map<String, String> additionalInfo = null;

        if (body.containsKey(META_ADDITIONAL_INFO_FIELD)) {
            try {
                additionalInfo = (Map<String, String>) body.get(META_ADDITIONAL_INFO_FIELD);
            } catch (ClassCastException e) {
                throw new IllegalArgumentException("Invalid type for field [" + META_ADDITIONAL_INFO_FIELD + "]", e);
            }
        }
```

Reply:
Essentially what I am trying to do is: if we have an unknown parameter in the request body we want to throw an error. So the default case here throws an IllegalArgumentException. So if there is an unknown field it will be caught in the loop and an exception is thrown in the default case.

With the method you suggested, we are reading the known/expected parameters but we won't be dealing with unknown/bad parameters.

Reply:
Do we need to throw the exception? If we get our desired value from the request, why do we need to throw the exception? Usually we don't do that. Customer can add numerous amount of items in the payload request, but we only care for the values what we need. We just discard other values, we don't need to throw exception for that.

Reply:
I was trying to cater to this part of the [issue](https://github.com/opensearch-project/ml-commons/issues/3250#:~:text=when%20sending%20fields%20that%20is%20not%20defined%20in%20the%20mapping%2C%20it%20also%20creates%20new%20messages%20with%20fields%20in%20null%20values.%20It%20doesn%27t%20throw%20exception.)

@mingshl, would love to hear your thoughts about this.

Reply:
I personally believe throwing exceptions is the way to go. Because we can enforce a strict contract and if the customers make an unintentional mistake perhaps a typo, they can quickly realize their error and remedy right then,